main/utils.go

```diff
@@ -269,3 +270,65 @@ func getStdPipesAndTelemetry(lg ExtensionLogger, logDir string, runErr error) {
 	isSuccess := runErr == nil
 	telemetry("output", msgTelemetry, isSuccess, 0)
 }
+
+func updateAssignment(assignmentName string, contentHash string) {
+    if( assignmentName == "" || contentHash == "" ) {
+        return
+    }
+
+    dscConfigFilePath := "/var/lib/GuestConfig/dsc/dsc.config"
+
+    configJsonFile, err := os.Open(dscConfigFilePath)
+    if err != nil {
+        dscConfigFolderPath := "/var/lib/GuestConfig/dsc"
+        os.MkdirAll(dscConfigFolderPath, os.ModePerm)
+
+        // New dsc.config file
+        firstAssignmentString := "{\"Assignments\": [ { \"name\": \"" + assignmentName + "\", \"contentHash\": \"" + contentHash + "\" }]}"
+        var firstAssignment map[string]interface{}
+        json.Unmarshal([]byte(firstAssignmentString ), &firstAssignment)
+        firstAssignmentJson, _ := json.Marshal(firstAssignment)
+        err = ioutil.WriteFile(dscConfigFilePath, firstAssignmentJson, 0644)
```

Comment:
>0644 [](start = 71, length = 4)

what is this number .. could you define a const for it ?

Reply:
this means owner have permission to read & write and other can only read. defined a const for this

---
In reply to: [296053478](https://github.com/Azure/Guest-Configuration-Extension/pull/87#discussion_r296053478) [](ancestors = 296053478)